legend-engine-language-pure-dsl-service-execution/src/main/java/org/finos/legend/engine/language/pure/dsl/service/execution/AbstractServicePlanExecutor.java

```diff
@@ -56,7 +51,7 @@
 
     private final String servicePath;
     private final ExecutionPlan plan;
-    private final PlanExecutor executor;
+    private PlanExecutor executor;
```

Comment:
This class really should be immutable and absolutely must be thread safe. What's the use case for changing this at runtime?